File: backend/strategy-game-common-data/src/main/kotlin/io/github/smiley4/strategygame/backend/commondata/Game.kt
package io.github.smiley4.strategygame.backend.commondata

data class Game(
    val id: Id,
    val name: String,
    val creationTimestamp: Long,
    var turn: Int,
    val players: PlayerContainer
) {
    @JvmInline
    value class Id(val value: String)
}

File: frontend/js/src/models/misc/game.ts
import {BrandedId} from "../../common/brandedId";

/**
 * Data about a game.
 */
export interface Game {
	id: Game.Id,
	name: string,
	creationTimestamp: number,
	currentTurn: number
}

export namespace Game {

	export type Id = BrandedId<string, "GameId">;

}